src/Adapter/MSTest.TestAdapter/Execution/TestAssemblyInfo.cs

```diff
@@ -234,4 +234,58 @@ public void RunAssemblyInitialize(TestContext testContext)
             }
         }
     }
+
+    /// <summary>
+    /// Calls the assembly cleanup method in a thread-safe.
+    /// </summary>
+    /// <remarks>
+    /// It is a replacement for RunAssemblyCleanup but as we are in a bug-fix version, we do not want to touch
+    /// public API and so we introduced this method.
+    /// </remarks>
+    internal void ExecuteAssemblyCleanup()
+    {
+        if (AssemblyCleanupMethod == null)
```

Comment:
should you double check inside the lock?

Reply:
I will keep as-is as I am only duplicating the existing method but I will keep a note to try to improve concurrency on these classes.